src/js/blocks/hero/block.js

```diff
@@ -0,0 +1,379 @@
+							<div class="poster">
+								<MediaUpload
+									onSelect={ media =>
+										{
+											setAttributes( {
+												alt: media.alt,
+												img: media.url,
+												mediaID: media.id
+											});
+										}
+									}
+									type="image"
+									value={ attributes.mediaID }
+									render={ ({ open }) => getImageButton( open ) }
+								/>
+							</div>
+							<div class="cl-hero-text overlay">
+								<div class="block">
+									<h1><PlainText
+										onChange={ content => setAttributes({ headline: content }) }
+										value={ attributes.headline }
+										placeholder={ __( 'Your hero title' ) }
+										keepPlaceholderOnFocus={ true }
+									/></h1>
+									<p class="subhead"><RichText
+										onChange={ content => setAttributes({ subhead: content }) }
+										value={ attributes.subhead }
+										placeholder={ __( 'Your hero subtitle' ) }
+										keepPlaceholderOnFocus={ true }
+										className="subhead"
+									/></p>
+									<span class="cl-button">
+									<PlainText
+										onChange={ content => setAttributes({ button: content }) }
+										value={ attributes.button }
+										placeholder={ __( 'Your button text' ) }
+										keepPlaceholderOnFocus={true}
+									/></span>
+									{ meta }
+								</div>
+							</div>
+						</div>
+					</div>
+				</div>
+			);
+		};
+
+		// Generate block controls for alignment, etc
+		const createBlockControls = () => {
+			return (
+				<BlockControls key="controls">
+					<BlockAlignmentToolbar
+						value={ attributes.alignment }
+						onChange={ content => setAttributes({ alignment: content }) }
+					/>
+
+					{ !! attributes.img && (
+					<MediaUploadCheck>
+						<Toolbar>
+							<MediaUpload
+								onSelect={ media =>
+									{
+										setAttributes({
+											alt: media.alt,
+											img: media.url,
+											mediaID: media.id
+										});
+									}
+								}
+								allowedTypes={ ALLOWED_MEDIA_TYPES }
+								value={ attributes.mediaID }
+								render={ ( { open } ) => (
+									<IconButton
+										className="components-toolbar__control"
+										label={ __( 'Edit media' ) }
+										icon="edit"
+										onClick={ open }
+									/>
+								) }
+							/>
+						</Toolbar>
+					</MediaUploadCheck>
+					) }
+				</BlockControls>
+			);
+		};
+
+		// Generate sidebar inspector controls for other custom attributes
+		const createInspectorControls = () => {
+			return (
+				<InspectorControls>
+					<PanelBody>
+						<PanelRow>
+							<BaseControl
+								label={ __( 'Format' ) }
+							>
+								<ButtonGroup aria-label={ __( 'Hero Format' ) }>
+									{ [ 'default', 'fullwidth', 'super' ].map( ( value ) => {
+										const capitalizedValue = value.charAt( 0 ).toUpperCase() + value.slice( 1 );
+										const key = ( 'default' === value ) ? '' : value;
+										const format = ( undefined == attributes.format ) ? '' : attributes.format;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unexpected use of undefined. (no-undefined)](https://app.codacy.com/manual/uriweb/uri-component-library/pullRequest?prid=4757242)